src/App.js

```diff
@@ -11,19 +42,56 @@ class App extends React.Component {
       cardDeck: makeShuffledDeck(),
       // currCards holds the cards from the current round
       currCards: [],
+      gameStarted: false,
+      winner: "",
+      player1Score: 0,
+      player2Score: 0,
+      roundsLeft: maxRounds,
     };
   }
 
+  // Deal two cards - first card is for Player 1 and second card is for Player 2
   dealCards = () => {
+    // this.state.cardDeck.pop() modifies this.state.cardDeck array
+    const newCurrCards = [this.state.cardDeck.pop(), this.state.cardDeck.pop()];
+    let winner = "";
+    if (newCurrCards[0].rank > newCurrCards[1].rank) {
+      winner = "Player 1";
+    } else if (newCurrCards[0].rank < newCurrCards[1].rank) {
+      winner = "Player 2";
+    } else {
+      winner = "Nobody";
+    }
+
     this.setState((state) => ({
```

Comment:
Please note that this parameter is "previousState", offering you a snapshort of the previous state